src/main/java/spring/projects/e_commerce/website/service/ProductService.java

```diff
@@ -0,0 +1,85 @@
+package spring.projects.e_commerce.website.service;
+
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import spring.projects.e_commerce.website.dto.ProductDto;
+import spring.projects.e_commerce.website.dto.ProductUpdateDto;
+import spring.projects.e_commerce.website.entity.Product;
+import spring.projects.e_commerce.website.repository.ProductRepository;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class ProductService {
+    private final ProductRepository productRepository;
+    private final ModelMapper modelMapper;
+
+    public ResponseEntity<String> getAllProducts() {
+        List<Product> products = productRepository.findAll();
+        List<ProductDto> productDtos = new ArrayList<>();
+        for (Product product : products) {
+            productDtos.add(modelMapper.map(product, ProductDto.class));
+        }
+        if (products.isEmpty()) {
+            return ResponseEntity.badRequest().body("No products found");
+        } else {
+            return ResponseEntity.ok().body(productDtos.toString());
+        }
+    }
+
+    public ResponseEntity<String> addProduct(ProductDto productDto) {
```

Comment:
Add javakarta validation in ProductDto class, it`ll validate by itself. No need for extra check